dgraphtest/json.go

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2023 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dgraphtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/crc64"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func PollTillPassOrTimeout(gcli *GrpcClient, query, want string, timeout time.Duration) error {
+	ticker := time.NewTimer(requestTimeout)
+	defer ticker.Stop()
+	for {
+		resp, err := gcli.Query(query)
+		if err != nil {
+			return errors.Wrap(err, "error while query")
+		}
+
+		err = CompareJSON(want, string(resp.Json))
+		if err == nil {
+			return nil
+		}
+		select {
+		case <-ticker.C:
+			return err
+		default:
+			time.Sleep(waitDurBeforeRetry)
+		}
+	}
+}
+
+// CompareJSON compares two JSON objects (passed as strings).
+func CompareJSON(want, got string) error {
```

Comment:
Was `require.JSONEq(t, expected, actual)` not working for us? It's supposed to do all the work that you have just coded up.

Reply:
It doesn't print the diff as nicely as this function does.